### ISSUE TITLE
Implement rerank endpoint

### ISSUE THREAD

**Issue by miyako** [miyako/onnx-genai#2]
## Type A

Run rerank model

## Type B

Run ColBERT model

**Comment by miyako** [comment]
## Models to test with ONNX Runtime

- [x]  [`BAAI/bge-reranker-v2-m3`](https://huggingface.co/BAAI/bge-reranker-v2-m3) `xlm-roberta` `1 Logit`
- [ ]  [`cross-encoder/mmarco-mMiniLMv2-L12-H384-v1`](https://huggingface.co/cross-encoder/mmarco-mMiniLMv2-L12-H384-v1) `xlm-roberta`  `2 Logits`
- [ ]   [`mixedbread-ai/mxbai-rerank-xsmall-v1`](https://huggingface.co/mixedbread-ai/mxbai-rerank-xsmall-v1) `bert`
- [ ]  [`cross-encoder/ms-marco-MiniLM-L6-v2`](https://huggingface.co/cross-encoder/ms-marco-MiniLM-L6-v2) `bert`
- [ ]  [`ibm-granite/granite-embedding-reranker-english-r2`](https://huggingface.co/ibm-granite/granite-embedding-reranker-english-r2) `modernbert`
